File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/javascript_solutions/LCP 67-KnLfVT.js
var decorateTree = function(root) {
    if (!root) {
        return null;
    }

    function dfs(node) {
        if (!node.left && !node.right) {
            node.val = 1;
            return 1;
        }

        let leftVal = 0;
        if (node.left) {
            leftVal = dfs(node.left);
        }

        let rightVal = 0;
        if (node.right) {
            rightVal = dfs(node.right);
        }

        node.val = leftVal + rightVal;
        return node.val;
    }

    dfs(root);
    return root;
};

File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/python3_solutions/LCP 67-KnLfVT.py
from typing import Optional

class Solution:
    def decorateTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        MOD = 10**9 + 7

        def dfs(node: Optional[TreeNode]):
            if not node:
                return

            dfs(node.left)
            dfs(node.right)

            left_val = node.left.val if node.left else 0
            right_val = node.right.val if node.right else 0

            node.val = (node.val + left_val + right_val) % MOD

        dfs(root)
        return root